File: app/src/main/java/com/challenge/fidoreader/fagment/FingerItemFragment.java
package com.challenge.fidoreader.fagment;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.challenge.fidoreader.R;

public class FingerItemFragment extends LinearLayout {

    TextView textview;
    ImageView imageView;

    public FingerItemFragment(Context context) {
        super(context);
        init(context);
    }

    private void init(Context context) {
        LayoutInflater inflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater.inflate(R.layout.fragment_finger_list, this, true);
        textview = findViewById(R.id.textViewfp);
        imageView = findViewById(R.id.imageViewfp);
    }

    public FingerItemFragment(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public void setName(String name){
        textview.setText(name);
    }

    public void setImage(int resid){
        imageView.setImageResource(resid);
    }


}

File: app/src/main/java/com/challenge/fidoreader/fagment/FingerItemFragment.kt
package com.challenge.fidoreader.fagment


import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.challenge.fidoreader.R


class FingerItemFragment : LinearLayout {
    var textview: TextView? = null
    var imageView: ImageView? = null

    constructor(context: Context) : super(context) {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        inflater.inflate(R.layout.fragment_finger_list, this, true)
        textview = findViewById(R.id.textViewfp)
        imageView = findViewById(R.id.imageViewfp)
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}

    fun setName(name: String?) {
        textview!!.text = name
    }

    fun setImage(resid: Int) {
        imageView!!.setImageResource(resid)
    }
}
